### ISSUE TITLE
`incorrect-version` rejects `1.0.0-alpha.1` and `1.0.0+build`, which SemVer defines

### ISSUE THREAD

**Issue by gemshrine** [objectionary/lints#1368]
`incorrect-version` reports two version strings that SemVer defines, while telling the author that SemVer is what it wants.

`src/main/resources/org/eolang/lints/metas/incorrect-version.xsl:18`:

```xml
<xsl:if test="$meta-head='version' and not(matches($meta-tail, '^\d+\.\d+\.\d+(-[a-zA-Z0-9-]+)?$|^\d+\.\d+-SNAPSHOT$'))">
```

and the message it produces:

```
The format of the +version meta is wrong: <tail> (SemVer expected instead)
```

Two shapes that [SemVer 2.0.0](https://semver.org/) spells out are refused by that pattern:

| `+version` | SemVer | the rule |
| --- | --- | --- |
| `0.0.1-alpha` | valid | accepted |
| `0.0.1-beta-1` | valid | accepted |
| `1.0.0-alpha.1` | valid — §9, dot-separated pre-release identifiers | **reported** |
| `1.0.0+20130313144700` | valid — §10, build metadata | **reported** |

The pre-release group is `[a-zA-Z0-9-]+`, which has no dot in it, so a numbered pre-release such as `-alpha.1` or `-rc.2` — the form SemVer itself uses in its examples — falls through. Build metadata is not in the pattern at all, in either alternative.

Verified as two packs under `incorrect-version`, each with `defects: 0`, both failing on master:

```
Tests run: 490, Failures: 2
  incorrect-version/probe-semver.yaml
  incorrect-version/probe-build.yaml
```

while `allows-good-versions.yaml` beside them keeps passing, so it is these two shapes specifically.

## Root cause

The pattern was written from the common cases rather than from the grammar. The second alternative, `^\d+\.\d+-SNAPSHOT$`, is a deliberate Maven exception, so the intent is "SemVer, plus the Maven two-part snapshot" — and the first alternative falls short of the SemVer half.

## Suggested fix

The official SemVer regular expression, with the Maven alternative kept:

```xml
<xsl:if test="$meta-head='version' and not(matches($meta-tail, '^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-((0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(\+([0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*))?$|^\d+\.\d+-SNAPSHOT$'))">
```

That also starts rejecting a leading zero in a numeric component (`01.2.3`), which SemVer forbids and the current pattern allows; if that is more strictness than wanted, `\d+` in the three places keeps today's behaviour.

Either way the two shapes above deserve packs, so the rule and the standard it names cannot drift apart again.

**Comment by 0crat** [comment]
@gemshrine Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+3830](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
